reports/pin/pin.qmd

```diff
@@ -15,23 +15,364 @@ format:
+  if (nchar(target_pin) != 14) {
+    stop(paste("Invalid PIN:", target_pin, "must be 14 characters long"))
+  }
+
+  if (!target_pin %in% assessment_card$meta_pin) {
+    stop(paste("Invalid PIN:", target_pin, "not found in assessment data"))
+  }
 }
+```
 
-if (!target_pin %in% assessment_card$meta_pin) {
-  stop("Invalid PIN: not found in assessment data")
+```{r _pin_URLs, results='asis'}
+base_url <- "https://www.cookcountyassessor.com/pin/"
+pin_values <- params$pin
+
+for (pin in pin_values) {
+  full_url <- paste0(base_url, pin)
+  cat(sprintf('<a href="%s">%s</a><br/>', full_url, full_url))
 }
 ```
 
-This is a stub report for the PIN `r target_pin` and run ID `r params$run_id`!
+```{r _pin_setup}
+noctua_options(cache_size = 10)
+con <- dbConnect(noctua::athena())
+
+pins_clause <- paste0("('", paste(params$pin, collapse = "', '"), "')")
+
+res_chars <- dbGetQuery(
+  conn = con,
+  sprintf("SELECT pin, year, pin10, pin_is_multicard, card, class, char_yrblt, char_bldg_sf, char_beds, char_fbath, char_hbath, char_air, char_type_resd, char_rooms FROM default.vw_card_res_char WHERE CAST(year AS INTEGER) >= %d AND pin IN %s", as.numeric(params$year) - 3, pins_clause)
+)
+
+pin_value <- dbGetQuery(
+  conn = con,
+  sprintf("SELECT * FROM default.vw_pin_value WHERE CAST(year AS INTEGER) >= %d AND pin IN %s", as.numeric(params$year) - 3, pins_clause)
+)
+
+pin_sale <- dbGetQuery(
+  conn = con,
+  sprintf("SELECT year, pin, num_parcels_sale, sv_outlier_type, doc_no, sale_date, sale_price FROM default.vw_pin_sale WHERE CAST(year AS INTEGER) >= %d AND pin IN %s", as.numeric(params$year) - 10, pins_clause)
+)
+
+hie_data_subset <- hie_data %>%
+  filter(hie_last_year_active >= params$year) %>%
+  mutate(hie = TRUE)
+```
+
+## Predicted Model Value
+
+```{r, _pin_predicted_model_value}
+assessment_pin %>%
+  mutate(township = ccao::town_convert(meta_township_code)) %>%
+  filter(meta_pin %in% params$pin) %>%
+  arrange(meta_pin) %>%
+  select(
+    "Township" = township,
+    "Final FMV" = pred_pin_final_fmv
+  ) %>%
+  mutate(`Final FMV` = scales::dollar(`Final FMV`)) %>%
+  kable() %>%
+  kable_styling(
+    "striped",
+    position = "left",
+    fixed_thead = TRUE,
+    html_font = "Consolas"
+  )
+```
+
+## Sale Data
+
+```{r _pin_topline_characteristics, results= 'asis'}
+assessment_pin %>%
+  filter(meta_pin %in% params$pin) %>%
+  left_join(pin_sale, by = c("meta_pin" = "pin")) %>%
+  arrange(meta_pin, desc(sale_date)) %>%
+  select(
+    "Sale Date" = sale_date,
+    "Sale Price" = sale_price,
+    "Document Number" = doc_no,
+    "Number of Parcels" = num_parcels_sale,
+    "Outlier Type" = sv_outlier_type
+  ) %>%
+  mutate(`Sale Price` = as.numeric(`Sale Price`)) %>%
+  mutate(`Sale Price` = dollar(`Sale Price`)) %>%
+  kable() %>%
+  kable_styling(
+    "striped",
+    position = "left",
+    fixed_thead = TRUE,
+    html_font = "Consolas"
+  )
+```
+
+## Characteristics
+
+```{r _pin_characteristics}
+res_chars %>%
```

Comment:
**issue:** The table itself is fine, but basically all the information in it can be pulled from `input/chars_data.parquet` rather than SQL.